scalafix-testkit/src/main/scala/scalafix/testkit/SemanticRuleSuite.scala

```diff
@@ -98,7 +100,7 @@ abstract class SemanticRuleSuite(
         .collectFirst { case f if f.isFile => f.readAllBytes }
         .map(new String(_))
         .getOrElse {
-          if (Patch.isOnlyLintMessages(patch)) obtained // linter
+          if (Patch.isAllAtomic(patch) && Patch.isOnlyLintMessages(patch)) obtained // linter
           else {
```

Comment:
I'm not familiar with this area but it seems that we need to implement this validation somewhere outside of this IF. Otherwise tests would fail with `java.lang.RuntimeException: Missing expected output file for test ...`, which is not the correct reason (see build results)

Reply:
I'm not sure where should I put the check here @olafurpg pointed around here.
To me it sounds like I should just skip all the code after line 87, can anyone help on this?

Reply:
Hey @joan38, sorry for not replying to your comment earlier.

I believe we can implement this validation after line 108. It seems fair to me to validate the output of rules after we know the test suite is properly set up (see validation in line 104). I think we can do something like:

```scala
if (!Patch.isAllAtomic(patch)) {
  sys.error(
    s"One or more rules used by test ${diffTest.filename} emit non-atomic " +
      "top level patches. To fix this, please use `.atomic` on individual or group " +
      "of patches that need to be atomically applied.")
}
```

This would fail a test when a non-atomic top level patch is detected. Please feel free to change the error message if you come up with something better.
It would be nice to print the exact rule name so that users could spot the faulty one straight away. However, I observed that when multiple rules are used by a given test, their names are concatenated with `+`, which wouldn't make the error message more helpful I suppose.

I hope it helps 😉